### PR TITLE
[To test on HDD] Write changesets to tx instead of mutation

### PR DESCRIPTION
**Pull request by AskAlexSharov**
I don't know - will "write changesets to tx instead of batch" speedup or slow down blocks execution on HDD

Expected:
- Eat 20% more RAM - because state keys in batch eating more RAM than keys of changesets
- fsync may get longer - because each batch will write more data

### CODE REVIEW

**Comment by AlexeyAkhunov** [comment]
I have tried this before, and it kind of worked (but not on LMDB), but the effect was not really noticeable. The downside noted by @yperbasis was that there could be inconsistencies between changesets and the current state at any time, and that would cause some retracing APIs fail

**Comment by AskAlexSharov** [comment]
"retracing APIs fail" - I do not write into DB, but do write into write transaction - any read-only API's will not see this changes until: 
1) flush mutation to writeTx 2) then commit writeTx. 

"effect was not really noticeable" - blocks processing speed on 500Mb mutation+this feature is equal to 8GB mutation without this feature (but of course - it's only useful for Genesis sync, not for new blocks processing). As a side-effect of this feature - we may reduce mutation size (max RAM requirements). Another side-effect - OS may start write immutable things (changesets) to disk - way earlier than we call `fsync()` - it may speedup `fsync`.

**Comment by AlexeyAkhunov** [comment]
Ok, I will test it out with my HDD sync

File: core/state/plain_state_writer.go [hunk 1]
```diff
@@ -165,19 +171,23 @@ func (w *PlainStateWriter) WriteChangeSets() error {
...
 		}
-		if err = w.db.Put(dbutils.PlainStorageChangeSetBucket, key, storageSerialized); err != nil {
+		if err = db.Put(dbutils.PlainStorageChangeSetBucket, key, storageSerialized); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 func (w *PlainStateWriter) WriteHistory() error {
```
Comment (AlexeyAkhunov) [inline core/state/plain_state_writer.go:181]: Do you think this function is used anywhere? I suspect it might be not

**Comment by AlexeyAkhunov** [comment]
Unfortunately, I did not do clean testing with comparing the full sync without this change and with this change, but instead running it to continue my HDD run. I am happy to merge it, please just check if we can remove `PlainStateWriter.WriteHistory`

**Comment by AskAlexSharov** [comment]
Why do you think it worth to merge? Did you see some positive impact? No problem with too long fsync?

**Comment by AskAlexSharov** [comment]
Why do you think it worth to merge? Did you see some positive impact? No problem with too long fsync?

**Comment by AlexeyAkhunov** [comment]
Yes, I think it is worth merging. Mutation is pretty useless for writing changeSets, it just introduces another round of copying. Thinking of that, you can also do with for `WriteReceipts`, to increase the impact :)

**Comment by AskAlexSharov** [comment]
`PlainStateWriter.WriteHistory` is used in `Genesis.CommitGenesisState`
`WriteReceipts` - yep, added.

**Pull request merged**